### ISSUE TITLE
Why we have WAF in spoke virtual network, when we can have it centrally in Hub Vnet

### ISSUE THREAD

**Issue by nitinkhandelwal26** [mspnp/aks-secure-baseline#60]
Hello Team,

In this reference architecture, we are having WAF in Spoke vnet

Can you please explain why to have this in Spoke vnet when we can manage it centrally in Hub Vnet. Any considerations on this?

![image](https://user-images.githubusercontent.com/10863548/91302225-c3edc500-e7c3-11ea-82e9-7ff73ff2d410.png)

Regards,

**Comment by ferantivero** [comment]
thanks for your inquiry @nitinkhandelwal26, although the Azure Application Gateway could've been deployed on another vNet (e.g. hub), in this reference implementation the desired is to get different instances of Azure Application Gateway per cluster, so that's the reason why it gets provisioned into the spoke. Alternatively, you could consider having a single Azure Application Gateway for all your clusters if that suits better in your solution, and in that scenario you might consider moving it to your hub.  

In other words, the WAF is meant to be cluster-specific in this reference implementation, so it is sit as close to the cluster as possible.

**Comment by ckittel** [comment]
To add on to the advice offered by @ferantivero, Generally speaking @nitinkhandelwal26, Microsoft recommends single-tenant clusters.  Meaning clusters that are context-bound to solve for a single purpose (application boundry), etc. This is in contrast to a cluster that holds many disperate workloads. This allows the application ops team to manage the lifecycle of their application as close to "end-to-end" as possible.  This means, if you think about what would "go away" if the application were to "go away" we want to group those azure resources as reasonably together as possible.  Now, more to your actual point -- following that same model, if App Gateway was a "shared" resource (in a spoke for example), then instead of the gateway going away, you'd have route configuration going away in that shared service.  That means the lifecycle of AppGateway is longer than the lifecycle of the cluster.  That's FINE to do, it just involves transferring ownership and setting up lifecycle/onboarding/offboarding policies/governance around that resource.  But it does give you "centralize management" -- so it's a tradeoff (as are so many things)!  If you wish to treat Application Gateway as a managed service offering for a section of your org, you're then responsible for all that means around that, and you decouple the "management" of it from your workload.

We like to, as a default stance, empower application ops teams to manage their own resources to the fullest extent.  We drew a line between the networking team (the foundational hub-spoke) and the workload team (everything IN the spoke).  But you're free to draw that line wherever it makes sense for your organization roles/responsibilities, cost constraints, and "blast radius" considerations for a faulty resource (or misconfiguration).

Just think about the birth/death (lifecycle) of those services, and "break out" those services into IaC templates that reflect that.  Just also be aware of how you make "cross customer (workload)" configuration to those central resources in a way that is manageable and redeployable in a recovery situation.

**Comment by nitinkhandelwal26** [comment]
Thank you so much for explanation
